packages/icons/package.json

```diff
@@ -2,8 +2,12 @@
     "name": "@dhis2/ui-icons",
     "version": "6.5.0",
     "description": "Icons used in the UI libs",
-    "main": "build/cjs/lib.js",
-    "module": "build/es/lib.js",
+    "main": "./build/cjs/index.js",
+    "module": "./build/es/index.js",
+    "exports": {
+        "import": "./build/es/index.js",
+        "require": "./build/cjs/index.js"
+    },
```

Comment:
@Mohammer5 I think you just want to update these (and the `entryPoints.lib` in `d2.config.js`) to `<base>/react/index.js` right?

Reply:
That produces the following output when building:

```
[build:icons] Build parameters: 
[build:icons]  - Mode : production
[build:icons] [WARNING] Invalid "main" field in package.json, expected "./build/cjs/index.js" (got "./build/cjs/react/index.js") 
[build:icons] [WARNING] Invalid "module" field in package.json, expected "./build/es/index.js" (got "./build/es/react/index.js") 
[build:icons] [WARNING] Invalid "pkg.exports.import" field in package.json, expected "./build/es/index.js" (got "./build/es/react/index.js") 
[build:icons] [WARNING] Invalid "pkg.exports.require" field in package.json, expected "./build/cjs/index.js" (got "./build/cjs/react/index.js") 
```

Reply:
Which causes the build command to become stale

Reply:
There is a bug here in the app-scripts: https://github.com/dhis2/app-platform/blob/master/cli/src/lib/validators/validatePackageExports.js#L42

The `entrypointBasename` at this point is `index.js`, and then when the expected exports `expectedESMExport` and `expectedCJSExport` are constructed, it is assumed that the `entrypointBasename` is always under `build/esm/index.js`.

In this case, the constructed path should be `build/esm/react/index.js` so the logic in app-scripts needs to be adjusted.

Reply:
Or, we say that there must be a `index.js` in the source directory, and don't support nested entry-points under `<source>/<other>/index.js`.

Reply:
I guess the `src/index.js` workaround in this PR should be ok for now, I'll create an issue (if you haven't done it already)

Reply:
https://github.com/dhis2/app-platform/pull/new/fix-validation-exports